docs/dev/table/hive/index.md

```diff
@@ -263,3 +259,5 @@ The following limitations in Hive's data types impact the mapping between Flink
 * Flink's `TIMESTAMP_WITH_TIME_ZONE` is not supported by Hive
 * Flink's `TIMESTAMP_WITH_LOCAL_TIME_ZONE` is not supported by Hive
 * Flink's `MULTISET` is not supported by Hive
+* Hive's `UNION` type is not supported
```

Comment:
I'll make these couple changes of data types in a different PR so this PR can focus on Hive source/sink. Thanks!